pkg/versioninfo/variables.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 The ClusterLink Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package versioninfo uses Go's runtime/debug to set executable revision information.
+package versioninfo
+
+import (
+	_ "embed"
+	"strings"
+	"time"
+)
+
+// @TODO
+// The placement of version.txt inside this package is not ideal.
+// Unfortunately Go does not allow ".." in embed paths. A possible workaround is to
+// use "go:generate" comment to copy the file from the root to this location (and adding
+// the local version to .gitignore...)
+// An alternative would be to use a go:generate comment with "git describe" to find
+// the latest tag reachable from the current revision..
+var (
+	//go:embed version.txt
+	semver string
+	// SemVer will be the version specified in version.txt file
+	SemVer = strings.TrimSpace(semver)
```

Comment:
You don't prefer to do it in the init functions?

Reply:
It is not needed - this way it is done at build time, not runtime.